__integrations__/models/profile.test.ts

```diff
@@ -153,8 +152,8 @@ describe("Models |> Profile", () => {
     // update document
     const updates = {
       servicePreferencesSettings: {
-        mode: ServicesPreferencesModeEnum.AUTO,
-        version: 1 as NonNegativeInteger
+        mode: ServicesPreferencesModeEnum.AUTO as const,
```

Comment:
@BurnedMarshal please note this counter-intuitive behaviour: 

as we didn't define `mode` as `ServicesPreferencesModeEnum`, but we tell it to be either 
```
ServicesPreferencesModeEnum.AUTO | ServicesPreferencesModeEnum.MANUAL
``` 
or 
```
ServicesPreferencesModeEnum.LEGACY
```
 the type system cannot infer the union we defined is complete in regards of `ServicesPreferencesModeEnum`.

That is, as `ServicesPreferencesModeEnum.AUTO` is of type `ServicesPreferencesModeEnum`, it cannot be assigned to `mode` as the latter is defined with a possibly narrower set of values (if not even considered to be a totally different type).

To overcome such, we must tell the specific value we're assigning to be considered `as const`.

Reply:
What if we export `as ServicePreferencesSettings` and we use 
```
const updates = {
      servicePreferencesSettings: {
        mode: ServicesPreferencesModeEnum.AUTO,
        version: 1
      } as ServicePreferencesSettings
    };
```

Reply:
or `as Profile["servicePreferencesSettings"]` without any new export